src/renderers/noop/ReactNoopEntry.js

```diff
@@ -178,89 +178,105 @@ var SharedHostConfig = {
-      type: string,
-      oldProps: Props,
-      newProps: Props,
-      internalInstanceHandle: Object,
-      keepChildren: boolean,
-      recyclableInstance: null | Instance,
-    ): Instance {
-      const clone = {
-        id: instance.id,
-        type: type,
-        children: keepChildren ? instance.children : [],
-        prop: newProps.prop,
-      };
-      Object.defineProperty(clone, 'id', {
-        value: clone.id,
-        enumerable: false,
-      });
-      return clone;
-    },
+var MutationHostConfig = {
+  commitMount(instance: Instance, type: string, newProps: Props): void {
+    // Noop
+  },
+
+  commitUpdate(
+    instance: Instance,
+    updatePayload: Object,
+    type: string,
+    oldProps: Props,
+    newProps: Props,
+  ): void {
+    instance.prop = newProps.prop;
+  },
+
+  commitTextUpdate(
+    textInstance: TextInstance,
+    oldText: string,
+    newText: string,
+  ): void {
+    textInstance.text = newText;
+  },
+
+  appendChild: appendChild,
+  appendChildToContainer: appendChild,
+  insertBefore: insertBefore,
+  insertInContainerBefore: insertBefore,
+  removeChild: removeChild,
+  removeChildFromContainer: removeChild,
+
+  resetTextContent(instance: Instance): void {},
+};
+
+var PersistenceHostConfig = {
+  cloneInstance(
+    instance: Instance,
+    updatePayload: null | Object,
+    type: string,
+    oldProps: Props,
+    newProps: Props,
+    internalInstanceHandle: Object,
+    keepChildren: boolean,
+    recyclableInstance: null | Instance,
+  ): Instance {
+    const clone = {
+      id: instance.id,
+      type: type,
+      children: keepChildren ? instance.children : [],
+      prop: newProps.prop,
+    };
+    Object.defineProperty(clone, 'id', {
+      value: clone.id,
+      enumerable: false,
+    });
+    return clone;
+  },
 
-    cloneContainer(
-      container: Container,
-      recyclableContainer: Container,
-    ): Container {
-      return {rootID: container.rootID, children: []};
-    },
+  cloneContainer(
+    container: Container,
+    recyclableContainer: Container,
+  ): Container {
+    return {rootID: container.rootID, children: []};
+  },
 
-    appendInititalChildToContainer(
-      parentInstance: Container,
-      child: Instance | TextInstance,
-    ) {
-      parentInstance.children.push(child);
-    },
+  appendInititalChildToContainer(
+    parentInstance: Container,
+    child: Instance | TextInstance,
+  ) {
+    parentInstance.children.push(child);
+  },
 
-    finalizeContainerChildren(container: Container): void {},
+  finalizeContainerChildren(container: Container): void {},
 
-    replaceContainer(oldContainer: Container, newContainer: Container): void {
-      rootContainers.set(oldContainer.rootID, newContainer);
-    },
+  replaceContainer(oldContainer: Container, newContainer: Container): void {
+    rootContainers.set(oldContainer.rootID, newContainer);
   },
-});
+};
+
+// They are created lazily because only one can be created per test file.
```

Comment:
I thought you solved this? Why can't both modes be enabled?

Reply:
I think it's just because I would have to enable persistent reconciler for every test using Noop.
I could create only the persistent one lazily.

Reply:
Or I could enable it everywhere though it feels lazy.

Reply:
Why not just enable it in all tests by changing the default mock?

Reply:
Want to keep ReactDOM test paths more realistic.